YogaKit/YKLayout.h

```diff
@@ -0,0 +1,97 @@
+/**
+ * Copyright (c) 2014-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ */
+
+#import "YKEnums.h"
+
+@interface YKLayout : NSObject
+
+/**
+ The property that decides if we should include this view when calculating layout. Defaults to YES.
+ */
+@property (nonatomic, setter=setIncluded:) BOOL isIncluded;
+
+/**
+ The property that decides during layout/sizing whether or not yk_* properties should be applied. Defaults to NO.
+ */
+@property (nonatomic, setter=setEnabled:) BOOL isEnabled;
+
+@property (nonatomic) YKDirection direction;
+@property (nonatomic) YKFlexDirection flexDirection;
+@property (nonatomic) YKJustify justifyContent;
+@property (nonatomic) YKAlign alignContent;
+@property (nonatomic) YKAlign alignItems;
+@property (nonatomic) YKAlign alignSelf;
+@property (nonatomic) YKPositionType positionType;
+@property (nonatomic) YKWrap flexWrap;
+
+@property (nonatomic) CGFloat flexGrow;
+@property (nonatomic) CGFloat flexShrink;
+@property (nonatomic) CGFloat flexBasis;
+
+@property (nonatomic) CGFloat positionLeft;
+@property (nonatomic) CGFloat positionTop;
+@property (nonatomic) CGFloat positionRight;
+@property (nonatomic) CGFloat positionBottom;
+@property (nonatomic) CGFloat positionStart;
+@property (nonatomic) CGFloat positionEnd;
+@property (nonatomic) CGFloat positionHorizontal;
```

Comment:
Personally, I find this makes the API much more confusing. There are multiple properties that essentially do the same thing. I could set `positionLeft` and `positionRight` or just simply set `positionHorizontal`. To me, it would be much more idiomatic to Objective-C and Swift if we did something like this.

```objc
@property (nonatomic, readwrite, assign) UIEdgeInsets position;
```

or

```objc
- (void)setMargin:(CGFloat)margin forEdges:(YKEdges)edges;
```
where YKEdges is an NSOption.

I would apply this comment to `margin` and `padding` too. cc: @emilsjolander